File: mobile/android/focus-android/app/src/main/java/org/mozilla/focus/customtabs/CustomTabsService.java
/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.focus.customtabs;

import android.app.Service;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.os.IBinder;
import android.support.customtabs.ICustomTabsCallback;
import android.support.customtabs.ICustomTabsService;

import java.util.List;

public class CustomTabsService extends Service {
    @Override
    public IBinder onBind(Intent intent) {
        return new ICustomTabsService.Stub() {
            @Override
            public boolean warmup(long flags) {
                return true;
            }

            @Override
            public boolean newSession(ICustomTabsCallback callback) {
                return true;
            }

            @Override
            public boolean mayLaunchUrl(ICustomTabsCallback callback, Uri url, Bundle extras, List<Bundle> otherLikelyBundles) {
                return true;
            }

            @Override
            public Bundle extraCommand(String s, Bundle bundle) {
                return null;
            }

            @Override
            public boolean updateVisuals(ICustomTabsCallback callback, Bundle bundle) {
                return false;
            }

            @Override
            public boolean requestPostMessageChannel(ICustomTabsCallback iCustomTabsCallback,
                                                     Uri uri) {
                return false;
            }

            @Override
            public int postMessage(ICustomTabsCallback iCustomTabsCallback, String s,
                                   Bundle bundle) {
                return 0;
            }

            @Override
            public boolean validateRelationship(ICustomTabsCallback iCustomTabsCallback, int i, Uri uri, Bundle bundle) {
                return false;
            }
        };
    }
}

File: mobile/android/android-components/components/browser/session/src/main/java/mozilla/components/browser/session/tab/CustomTabsService.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package mozilla.components.browser.session.tab

import android.app.Service
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.os.IBinder
import android.support.customtabs.ICustomTabsCallback
import android.support.customtabs.ICustomTabsService

class CustomTabsService : Service() {
    override fun onBind(intent: Intent): IBinder? {
        return object : ICustomTabsService.Stub() {
            override fun warmup(flags: Long): Boolean = true
            override fun newSession(cb: ICustomTabsCallback) = true
            override fun mayLaunchUrl(cb: ICustomTabsCallback, url: Uri, extras: Bundle, bundles: List<Bundle>) = true
            override fun extraCommand(s: String, bundle: Bundle): Bundle? = null
            override fun updateVisuals(cb: ICustomTabsCallback, bundle: Bundle) = false
            override fun requestPostMessageChannel(cb: ICustomTabsCallback, uri: Uri): Boolean = false
            override fun postMessage(cb: ICustomTabsCallback, s: String, bundle: Bundle): Int = 0
            override fun validateRelationship(cb: ICustomTabsCallback, i: Int, uri: Uri, bundle: Bundle) = false
        }
    }
}
